### PR TITLE
Add database and external span attributes to correlate to metric data

### PR DESCRIPTION
**Pull request by sdaubin**
_Before contributing, please read our [contributing guidelines](https://github.com/newrelic/newrelic-java-agent/blob/main/CONTRIBUTING.md) and [code of conduct](https://github.com/newrelic/.github/blob/main/CODE_OF_CONDUCT.md)._

### Overview

This adds attributes to Spans so that we can correlate aggregate database and external metric back to their related client spans.

 * db.system
 * db.operation
 * db.collection
 * server.address
 * server.port

https://opentelemetry.io/docs/specs/otel/trace/semantic_conventions/http/
https://opentelemetry.io/docs/specs/otel/trace/semantic_conventions/database/

### Related Github Issue
Include a link to the related GitHub issue, if applicable

### Testing
I added unit tests to cover the new attributes.  I can a couple of local apps with a local build and verified that the attributes were correctly added.

### Checks

[ X ] Are your contributions backwards compatible with relevant frameworks and APIs?
[ ] Does your code contain any breaking changes? Please describe. 
[ ] Does your code introduce any new dependencies? Please describe.

### CODE REVIEW

File: newrelic-agent/src/main/java/com/newrelic/agent/service/analytics/SpanEventFactory.java [hunk 1]
```diff
@@ -215,42 +213,52 @@ public SpanEventFactory setHttpStatusText(String statusText) {
...
-        builder.putIntrinsic("peer.hostname", host);
+        builder.putIntrinsic("db.system", component);
         return this;
     }
 
     // datastore parameter
     public SpanEventFactory setAddress(String hostName, String portPathOrId) {
         if (portPathOrId != null && hostName != null) {
             String address = MessageFormat.format("{0}:{1}", hostName, portPathOrId);
-            builder.putIntrinsic("peer.address", address);
```
Comment (sdaubin) [inline newrelic-agent/src/main/java/com/newrelic/agent/service/analytics/SpanEventFactory.java:238]: `peer.address` is not intrinsic in the agent specs

File: newrelic-agent/src/main/java/com/newrelic/agent/service/analytics/SpanEventFactory.java [hunk 2]
```diff
@@ -219,42 +217,52 @@ public SpanEventFactory setHttpStatusText(String statusText) {
...
     // datastore parameter
     public SpanEventFactory setDatabaseName(String databaseName) {
-        builder.putIntrinsic("db.instance", databaseName);
+        builder.putAgentAttribute("db.instance", databaseName);
         return this;
     }
 
     // datastore parameter
     public SpanEventFactory setDatastoreComponent(String component) {
-        builder.putIntrinsic("component", component);
```
Comment (obenkenobi) [inline newrelic-agent/src/main/java/com/newrelic/agent/service/analytics/SpanEventFactory.java:228]: This may break custom dashboards used by customers

**Comment by obenkenobi** [comment]
This PR removes the attributes:

- `component`
- `peer.hostname`
- `db.collection`

This has caused the following unit tests to fail:

-`com.newrelic.agent.tracers.DefaultSqlTracerClmTest`
-`com.newrelic.agent.tracers.DefaultSqlTracerTest`
-`com.newrelic.agent.tracers.DefaultTracerClmTest`
-`com.newrelic.agent.tracers.DefaultTracerTest`

Why were these attributes were removed? Wouldn't that cause breaking changes with the agent assuming a customer is utilizing an NRQL query with these attributes in mind?

**Comment by sdaubin** [comment]
@obenkenobi I added `peer.hostname` back.  I'd like to remove that eventually, but we need to clean up some product references first.

**Comment by sdaubin** [comment]
@obenkenobi I reverted `db.sql.table` back to `db.collection` in this PR and in the spec PR.

**Review by obenkenobi — APPROVED** [review]

**Comment by obenkenobi** [comment]
I approved the PR but added a duplicate PR to resolve a merge conflict. https://github.com/newrelic/newrelic-java-agent/pull/1525/ 

If this gets merged, I will close this one out.

**Pull request merged**